### ISSUE TITLE
Custom JSON Field causes TypeError When Searching Subset of Filters

### ISSUE THREAD

**Issue by sirtux** [nautobot/nautobot#1678]
### Environment
* Python version: 3.7
* Nautobot version: 1.3.1

### Steps to Reproduce
1. Create a custom json field assigned to tenant, without setting a default value
2. Have a tenant-group with a tenant in it
3. Open http://host/tenancy/tenants/?group=tenant-test-group
-> Error
4. However, http://host/tenancy/tenant-groups/tenant-test-group/?tab=main works.

### Expected Behavior
List of tenants

### Observed Behavior
TypeError at /tenancy/tenants/
the JSON object must be str, bytes or bytearray, not NoneType

Traceback:
```
Environment:


Request Method: GET
Request URL: http://engineeringworkstation-alpha:8080/tenancy/tenants/?group=tenant-test-group

Django Version: 3.2.12
Python Version: 3.7.13
Installed Applications:
['django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'cacheops',
 'corsheaders',
 'django_filters',
 'django_jinja',
 'django_tables2',
 'django_prometheus',
 'mptt',
 'rest_framework',
 'social_django',
 'taggit',
 'timezone_field',
 'nautobot.core.apps.NautobotConstanceConfig',
 'nautobot.core',
 'django.contrib.admin',
 'django_celery_beat',
 'db_file_storage',
 'nautobot.circuits',
 'nautobot.dcim',
 'nautobot.ipam',
 'nautobot.extras',
 'nautobot.tenancy',
 'nautobot.users',
 'nautobot.utilities',
 'nautobot.virtualization',
 'django_rq',
 'nautobot.third_party.drf_spectacular',
 'drf_spectacular_sidecar',
 'graphene_django',
 'health_check',
 'health_check.storage',
 'django_extensions',
 'nautobot.core.apps.ConstanceDatabaseAppConfig',
 'django_ajax_tables',
 'debug_toolbar']
Installed Middleware:
['debug_toolbar.middleware.DebugToolbarMiddleware',
 'django_prometheus.middleware.PrometheusBeforeMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'nautobot.core.middleware.ExceptionHandlingMiddleware',
 'nautobot.core.middleware.RemoteUserMiddleware',
 'nautobot.core.middleware.ExternalAuthMiddleware',
 'nautobot.core.middleware.ObjectChangeMiddleware',
 'django_prometheus.middleware.PrometheusAfterMiddleware']


Template error:
In template /source/nautobot/core/templates/inc/search_panel.html, error at line 28
   the JSON object must be str, bytes or bytearray, not NoneType
   18 :                             <span class="input-group-btn">
   19 :                                 <button type="submit" class="btn btn-primary">
   20 :                                     <span class="mdi mdi-magnify" aria-hidden="true"></span>
   21 :                                 </button>
   22 :                             </span>
   23 :                         </div>
   24 :                     {% elif field|widget_type == 'checkboxinput' %}
   25 :                         <label for="{{ field.id_for_label }}">{{ field }} {{ field.label }}</label>
   26 :                     {% else %}
   27 :                         {{ field.label_tag }}
   28 :                          {{ field }} 
   29 :                     {% endif %}
   30 :                 </div>
   31 :             {% endfor %}
   32 :             <div class="text-right noprint">
   33 :                 <button type="submit" class="btn btn-primary">
   34 :                     <span class="mdi mdi-magnify" aria-hidden="true"></span> Apply
   35 :                 </button>
   36 :                 <a href="." class="btn btn-default">
   37 :                     <span class="mdi mdi-close-thick" aria-hidden="true"></span> Clear
   38 :                 </a>


Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/source/nautobot/utilities/views.py", line 94, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/source/nautobot/core/views/generic.py", line 265, in get
    return render(request, self.template_name, context)
  File "/usr/local/lib/python3.7/site-packages/django/shortcuts.py", line 19, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/usr/local/lib/python3.7/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
  File "/usr/local/lib/python3.7/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 170, in render
    return self._render(context)
  File "/usr/local/lib/python3.7/site-packages/django/test/utils.py", line 100, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.7/site-packages/django/test/utils.py", line 100, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/defaulttags.py", line 315, in render
    return nodelist.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/loader_tags.py", line 195, in render
    return template.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/usr/local/lib/python3.7/site-packages/django/test/utils.py", line 100, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/defaulttags.py", line 214, in render
    nodelist.append(node.render_annotated(context))
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/defaulttags.py", line 315, in render
    return nodelist.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 994, in render
    return render_value_in_context(output, context)
  File "/usr/local/lib/python3.7/site-packages/django/template/base.py", line 973, in render_value_in_context
    value = str(value)
  File "/usr/local/lib/python3.7/site-packages/django/utils/html.py", line 376, in <lambda>
    klass.__str__ = lambda self: mark_safe(klass_str(self))
  File "/usr/local/lib/python3.7/site-packages/django/forms/boundfield.py", line 34, in __str__
    return self.as_widget()
  File "/usr/local/lib/python3.7/site-packages/django/forms/boundfield.py", line 95, in as_widget
    value=self.value(),
  File "/usr/local/lib/python3.7/site-packages/django/forms/boundfield.py", line 130, in value
    data = self.field.bound_data(self.data, data)
  File "/usr/local/lib/python3.7/site-packages/django/forms/fields.py", line 1265, in bound_data
    return json.loads(data, cls=self.decoder)
  File "/usr/local/lib/python3.7/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '

Exception Type: TypeError at /tenancy/tenants/
Exception Value: the JSON object must be str, bytes or bytearray, not NoneType
```

**Comment by bryanculver** [comment]
Confirmed this issues is on more than just Tenants.

**Steps to Reproduce:**
1. Create JSON custom field, assign to any content type
2. Open list view of assigned content type
3. Perform search by either:
    - Providing a known filter name and value (such as `?name=x`) in URL
    - Using search widget then remove query param for JSON custom field from URL

**Comment by bryanculver** [comment]
This is a bug in Django which was fixed: https://github.com/django/django/pull/14558/files

However that fix has not been back ported to 3.2.

The fix: https://github.com/django/django/blob/main/django/forms/fields.py#L1363-L1364

Missing in 3.2.13: https://github.com/django/django/blob/3.2.13/django/forms/fields.py#L1264

Included in 4.X: https://github.com/django/django/blob/4.0/django/forms/fields.py#L1265-L1266

So for now, we need to catch the error until we upgrade to Django 4.

**Comment by bryanculver** [comment]
This also breaks pagination.
